pkgs/tools/misc/android-tools/default.nix

```diff
@@ -28,13 +28,30 @@ stdenv.mkDerivation rec {
     ./android-tools-kernel-headers-6.0.diff
   ];
 
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    sed -i 's/usb_linux/usb_osx/g' vendor/CMakeLists.{adb,fastboot}.txt
+    sed -i 's/libselinux libsepol/ /g;s#selinux/libselinux/include##g' vendor/CMakeLists.{fastboot,mke2fs}.txt
+    sed -z -i 's/add_library(libselinux.*selinux\/libsepol\/include)//g' vendor/CMakeLists.fastboot.txt
+    sed -i 's/e2fsdroid//g' vendor/CMakeLists.txt
+    sed -z -i 's/add_executable(e2fsdroid.*e2fsprogs\/misc)//g' vendor/CMakeLists.mke2fs.txt
+  '';
+
   nativeBuildInputs = [ cmake perl go ];
   buildInputs = [ protobuf zlib gtest brotli lz4 zstd libusb1 pcre2 fmt_7 ];
   propagatedBuildInputs = [ pythonEnv ];
 
   # Don't try to fetch any Go modules via the network:
   GOFLAGS = [ "-mod=vendor" ];
 
+  NIX_CFLAGS_COMPILE = lib.optionals stdenv.isDarwin [
+    "-D_DARWIN_C_SOURCE"
+  ];
+
+  NIX_LDFLAGS = lib.optionals stdenv.isDarwin [
+    "-framework CoreFoundation"
+    "-framework IOKit"
+  ];
```

Comment:
Was there a reason these went here instead of `buildInputs`?

Reply:
Because these frameworks are not explicitly linked and without which

```
Undefined symbols for architecture arm64:
  "_IOCreatePlugInInterfaceForService", referenced from:
      usb_open(int (*)(usb_ifc_info*), unsigned int) in usb_osx.cpp.o
```

Reply:
So upstream wasn't adding the frameworks to the compiler arguments, and thus it's required, right?

I ask since you usually have to add these frameworks as inputs, but it seems that they're being propagated from some other dependency.

Reply:
Yes. Should I add them to `buildInputs` in addition?

Reply:
No. Go propagates them, AFAICT, so let's just keep what works.